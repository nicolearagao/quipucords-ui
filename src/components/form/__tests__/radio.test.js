import React from 'react';
import { Radio } from '../radio';

describe('Radio Component', () => {
  it('should render a basic component', () => {
    const props = {};

    const component = renderComponent(<Radio {...props} />);
    expect(component).toMatchSnapshot('basic component');
  });

  it('should handle disabled, checked', () => {
    const props = {
      isDisabled: true
    };

    const component = renderComponent(<Radio {...props} />);
    expect(component).toMatchSnapshot('disabled');

    const componentActive = component.setProps({
      isDisabled: false
    });
    expect(componentActive).toMatchSnapshot('active');

    const componentChecked = component.setProps({
      isDisabled: false,
      isChecked: true
    });

    expect(componentChecked).toMatchSnapshot('checked');
  });

  it('should handle children as a label', () => {
    const props = {};
    const component = renderComponent(<Radio {...props}>lorem ipsum</Radio>);
    expect(component).toMatchSnapshot('children label radio');
  });

  it('should return an emulated onChange event', () => {
    const mockOnChange = jest.fn();
    const props = {
      onChange: mockOnChange
    };

    const component = renderComponent(<Radio {...props}>lorem ipsum</Radio>);
    const input = component.find('input');
    component.fireEvent.click(input, { currentTarget: {}, target: {} });

    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
