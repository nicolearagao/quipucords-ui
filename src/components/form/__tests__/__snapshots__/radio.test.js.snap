// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Component should handle children as a label: children label radio 1`] = `
<radio
  props="{
  "children": [
    {}
  ],
  "ariaLabel": "radio input",
  "checked": null,
  "className": "",
  "id": null,
  "isChecked": false,
  "isDisabled": false,
  "label": "",
  "name": null
}"
>
  <div
    class="pf-c-radio quipucords-form__radio "
  >
    <input
      aria-invalid="false"
      class="pf-c-radio__input"
      data-ouia-component-id="OUIA-Generated-Radio-5"
      data-ouia-component-type="PF4/Radio"
      data-ouia-safe="true"
      id="generatedid-"
      name="generatedid-"
      type="radio"
      value=""
    />
    <label
      class="pf-c-radio__label"
      for="generatedid-"
    >
      lorem ipsum
    </label>
  </div>
</radio>
`;

exports[`Radio Component should handle disabled, checked: active 1`] = `
<radio
  props="{
  "isDisabled": false,
  "ariaLabel": "radio input",
  "checked": null,
  "children": [],
  "className": "",
  "id": null,
  "isChecked": false,
  "label": "",
  "name": null
}"
>
  <div
    class="pf-c-radio pf-m-standalone quipucords-form__radio "
  >
    <input
      aria-invalid="false"
      aria-label="radio input"
      class="pf-c-radio__input"
      data-ouia-component-id="OUIA-Generated-Radio-3"
      data-ouia-component-type="PF4/Radio"
      data-ouia-safe="true"
      id="generatedid-"
      name="generatedid-"
      type="radio"
      value=""
    />
  </div>
</radio>
`;

exports[`Radio Component should handle disabled, checked: checked 1`] = `
<radio
  props="{
  "isDisabled": false,
  "ariaLabel": "radio input",
  "checked": null,
  "children": [],
  "className": "",
  "id": null,
  "isChecked": true,
  "label": "",
  "name": null
}"
>
  <div
    class="pf-c-radio pf-m-standalone quipucords-form__radio "
  >
    <input
      aria-invalid="false"
      aria-label="radio input"
      checked=""
      class="pf-c-radio__input"
      data-ouia-component-id="OUIA-Generated-Radio-4"
      data-ouia-component-type="PF4/Radio"
      data-ouia-safe="true"
      id="generatedid-"
      name="generatedid-"
      type="radio"
      value=""
    />
  </div>
</radio>
`;

exports[`Radio Component should handle disabled, checked: disabled 1`] = `
<radio
  props="{
  "isDisabled": true,
  "ariaLabel": "radio input",
  "checked": null,
  "children": [],
  "className": "",
  "id": null,
  "isChecked": false,
  "label": "",
  "name": null
}"
>
  <div
    class="pf-c-radio pf-m-standalone quipucords-form__radio "
  >
    <input
      aria-invalid="false"
      aria-label="radio input"
      class="pf-c-radio__input"
      data-ouia-component-id="OUIA-Generated-Radio-2"
      data-ouia-component-type="PF4/Radio"
      data-ouia-safe="true"
      disabled=""
      id="generatedid-"
      name="generatedid-"
      type="radio"
      value=""
    />
  </div>
</radio>
`;

exports[`Radio Component should render a basic component: basic component 1`] = `
<radio
  props="{
  "ariaLabel": "radio input",
  "checked": null,
  "children": [],
  "className": "",
  "id": null,
  "isChecked": false,
  "isDisabled": false,
  "label": "",
  "name": null
}"
>
  <div
    class="pf-c-radio pf-m-standalone quipucords-form__radio "
  >
    <input
      aria-invalid="false"
      aria-label="radio input"
      class="pf-c-radio__input"
      data-ouia-component-id="OUIA-Generated-Radio-1"
      data-ouia-component-type="PF4/Radio"
      data-ouia-safe="true"
      id="generatedid-"
      name="generatedid-"
      type="radio"
      value=""
    />
  </div>
</radio>
`;
